integration-cli/docker_cli_build_test.go

```diff
@@ -4375,6 +4375,27 @@ func (s *DockerSuite) TestBuildTimeArgHistoryExclusions(c *check.C) {
 	}
 }
 
+// Reported in moby/moby#32744
+func (s *DockerSuite) TestBuildTimeArgHistoryExclusionsMultipleBuiltins(c *check.C) {
+	imgName := "bldargtest"
+	dockerfile := fmt.Sprintf(`FROM busybox
+		RUN echo "Testing Build Args!"`)
+	buildImage(imgName,
+		cli.WithFlags("--build-arg", "http_proxy=foo",
+			"--build-arg", "https_proxy=bar",
+		),
+		build.WithDockerfile(dockerfile),
+	).Assert(c, icmd.Success)
```

Comment:
It is definitely a test for that use case, but we can't write a slow integration test for every possible input. The test suite would take multiple years to run.  

Unfortunately I don't think a unit test is going to be possible right now. I wrote one in #32773 after I refactored a lot of this code. Currently this code relies on runConfig being shared with the container config of a conatiner that was already created. It's effectively modifying the config of a container post-create, which to me is a huge violation of the boundaries between components.

After #32773 the updated cmd is passed as an argument to commit.

For now updating the existing test sounds good.